omegaconf/base.py

```diff
@@ -413,15 +416,71 @@ def _resolve_interpolation_from_parse_tree(
                 raise
             return None
 
-        assert resolved is not None
-        if isinstance(resolved, str):
-            # Result is a string: create a new StringNode for it.
-            return StringNode(
-                value=resolved,
-                key=key,
-                parent=parent,
-                is_optional=value._metadata.optional,
-            )
+        # If the output is not a Node already (e.g., because it is the output of a
+        # custom resolver), then we will need to wrap it within a Node.
+        must_wrap = not isinstance(resolved, Node)
+
+        if isinstance(value, AnyNode):
+            # If the node is untyped, we still need to validate against `None`
+            # in case the `optional` flag is set to `False`.
```

Comment:
AnyNode with optional False does not make a whole lot of sense to me.

In fact, it doesn't make sense to mypy as well:

1.py:
```python
from typing import *

x : Any = None
y : Optional[Any] = None
```
```shell
$ mypy --strict 1.py 
Success: no issues found in 1 source file
```

Do we have a use case for non optional Any node?
I just tested and it straightforward to remove the is_optional flag from AnyNode and always make it True.
http://paste.ubuntu.com/p/PRTffVb5vC/

Reply:
> AnyNode with optional False does not make a whole lot of sense to me.

I was also surprised, but I noticed that it happened (there were AnyNodes created with optional=False when running the test suite), so I thought they had to be supported. But if we can drop them as you suggested, I totally agree it's a better fix. I'll give it a shot.

Reply:
Oh, here is my comment.
Okay, so this can be fixed in the other PR addressing the optional cleanup.

Reply:
Done in #592

Reply:
Resolving as this was addressed in #592